sdk/nodejs/output.ts

```diff
@@ -312,6 +236,77 @@ To manipulate the value of this Output, use '.apply' instead.`);
             },
         });
     }
+
+    public get(): T {
+        throw new Error(`Cannot call '.get' during update or preview.
+To manipulate the value of this Output, use '.apply' instead.`);
+    }
+
+    public apply<U>(func: (t: T) => Input<U>): Output<U> {
+        const applied: Promise<OutputData<U>> =
+            Promise.all([this.promise(), this.isKnown, this.isSecret])
+                   .then(([value, isKnown, isSecret]) => applyHelperAsync<T, U>(value, isKnown, isSecret, func));
```

Comment:
all we do here is .all the underlying promise values, then pass along their data to a helper that will combine them into the new pieces of data for the new output.